src/main/scala/higherkindness/compendium/core/CompendiumService.scala

```diff
@@ -44,13 +50,30 @@ object CompendiumService {
       override def storeProtocol(
           id: ProtocolId,
           protocol: Protocol,
-          idlName: IdlName
-      ): F[ProtocolVersion] =
-        for {
-          _       <- ProtocolUtils[F].validateProtocol(protocol, idlName)
-          version <- MetadataStorage[F].store(id, idlName)
-          _       <- Storage[F].store(id, version, protocol)
-        } yield version
+          idlName: IdlName,
+          validation: Option[ValidationBool]
+      ): F[ProtocolVersion] = {
+
+        def storeWithValidation(
+            id: ProtocolId,
+            protocol: Protocol,
+            idlName: IdlName
+        ): F[ProtocolVersion] =
+          for {
+            _       <- ProtocolUtils[F].validateProtocol(protocol, idlName)
+            version <- MetadataStorage[F].store(id, idlName)
+            _       <- Storage[F].store(id, version, protocol)
+          } yield version
+
+        validation.fold(storeWithValidation(id, protocol, idlName)) {
+          case True => storeWithValidation(id, protocol, idlName)
+          case False =>
+            for {
+              version <- MetadataStorage[F].store(id, idlName)
+              _       <- Storage[F].store(id, version, protocol)
+            } yield version
+        }
```

Comment:
Yep, cats provide `flatTap` for that
```suggestion
        validation.getOrElse(ValidationBool.True) match {
          case ValidationBool.True => storeWithValidation(id, protocol, idlName)
          case ValidationBool.True => MetadataStorage[F].store(id, idlName).flatTap(Storage[F].store(id, _, protocol))
        }
```